opm/material/common/Tabulated1DFunction.hpp

```diff
@@ -475,8 +482,27 @@ class Tabulated1DFunction
                     lowerIdx = pivotIdx;
             }
 
-            assert(xValues_[lowerIdx] <= x);
-            assert(x <= xValues_[lowerIdx + 1]);
+            if (xValues_[lowerIdx] > x || x > xValues_[lowerIdx + 1]) {
```

Comment:
For the scenarios that we are trying to handling, x is a `NaN`.  Then the behavior here is a little different from the original assertions trying to do. 

When `NaN` is involved, all (almost) the logic comparisons are `false`. As a result, in the original code, assertion failures will be triggered but in the new code in this PR, it will not throw. 

Luckily, the new PR https://github.com/OPM/opm-material/pull/534 complements the check for `NaN` values.

Reply:
or 
```c++
if ( !(xValues_[lowerIdx] <= x && x <= xValues_[lowerIdx + 1]) ) {

```
will also do the work.